packages/database/src/core/AppCheckTokenProvider.ts

```diff
@@ -30,17 +31,26 @@ import { warn } from './util/util';
  */
 export class AppCheckTokenProvider {
   private appCheck?: FirebaseAppCheckInternal;
+  private serverAppAppCheckToken?: string;
+  private appName: string;
   constructor(
-    private appName_: string,
+    app: FirebaseApp,
     private appCheckProvider?: Provider<AppCheckInternalComponentName>
   ) {
+    this.appName = app.name;
+    if (_isFirebaseServerApp(app) && app.settings.appCheckToken) {
+      this.serverAppAppCheckToken = app.settings.appCheckToken;
+    }
     this.appCheck = appCheckProvider?.getImmediate({ optional: true });
     if (!this.appCheck) {
       appCheckProvider?.get().then(appCheck => (this.appCheck = appCheck));
     }
   }
 
   getToken(forceRefresh?: boolean): Promise<AppCheckTokenResult> {
```

Comment:
if `forceRefresh && this.serverAppAppCheckToken` should this throw or otherwise indicate that an unsupported call was made?

Reply:
Just checking: is the only time that `forceRefresh` is set to `true` for the App Check token is when the service returns an `UNAUTHENTICATED` error?

Reply:
It looks like, in RTDB, it will be set to true on any failed request (status !== 'ok') to the auth or app check endpoints.

Reply:
I suppose that throwing here would then be valid, as it would reduce the burden on the service on subsequent requests, if the app attempts to continue to use the defunct App Check Token. WDYT @hsubox76 ?